featuresScore.java

```diff
@@ -0,0 +1,76 @@
+
+public class featuresScore {
+	public static final int COLS = 10;
+	public static final int ROWS = 21;
+	
+	//currently 21 features. To be increased in the future
+	private final int numFeatures = 21;
+	private double[] scores = new double[numFeatures];
+	private int[][] grids;
+	
+	private int[] colHeights = new int[COLS];
+	
+	public featuresScore(State s) {
+		grids = s.getField();
+		
+		//First 10 are colHeight
+		for (int i = 0; i < COLS; i++) {
+			scores[i] = checkColHeight(i);
+		}
+		
+		//The next 9 are adjHeight
+		for (int i = 10; i < 10 + COLS - 1; i++) {
+			scores[i] = checkDiffAdjColHeights(i);
+		}
+		
+		//max col height
+		scores[20] = checkMaxColHeight();
+		
+		scores[21] = checkNumWallHoles();
+	}
+	
+	public double getScore(int index) {
+		return scores[index];
+	}
+	
+	public int getNumberOfFeatures() {
+		return numFeatures;
+	}
+	
+	//Check from above
+	public int checkColHeight(int col) {
+		int count = 0;
+		while (grids[count][col] != 0) {
+			count++;
+		}
+		return count;
```

Comment:
Oh I know why alr. Because the comment says it counts from above, so I assume the topmost row is 0. So the comment is wrong...
Oh but then again, it will count the height wrongly if the column has a hole, so this is still wrong... And the comment is right, we should start counting from above to get the correct value :p